linode/instancereservedip/framework_models.go

```diff
@@ -0,0 +1,114 @@
+package instancereservedip
```

Comment:
Sorry for all the confusion surrounding this PR! 

It looks like the changes in #1598 would make this resource implementation redundant so it can probably be dropped from this PR 👍

Correct me if I'm wrong, but I think the scope of this PR can probably be narrowed to just the changes to the `linode_instance` since we decided against making changes to `linode_instance_ip`.

Reply:
No worries! The changes in #1598 handles the following operations of reserved IP -Create, Fetch, List. The changes in this PR is only restricted to creation of a linode with an existing reserved IP + assigning reserved IP to linodes. So I dont think there is an overlap here, but I maybe wrong.